build_runner/lib/src/server/server.dart

```diff
@@ -43,9 +51,14 @@ class AssetHandler {
 
   AssetHandler(this._reader, this._rootPackage);
 
-  Future<Response> handle(Request request) {
-    var pathSegments = request.url.pathSegments;
-    if (request.url.path.endsWith('/')) {
+  Future<Response> handle(Request request, String rootDir) {
+    var pathSegments = <String>[];
+    if (rootDir != null) {
```

Comment:
when would this be null? If we don't expect it to ever be null we could add an `assert` instead of a conditional.

Reply:
originally I had this as an optional arg, so ya I will send a pr to clean it up :)